### ISSUE TITLE
Invalid HTML is being generated

### ISSUE THREAD

**Issue by brc-dd** [algolia/docsearch#1491]
1. Go to https://docsearch.algolia.com/ and open the search box.
2. Here is a screenshot of the elements:
    ![image](https://user-images.githubusercontent.com/40380293/183374863-35be6634-215d-4231-84c2-1d53d0833c96.png)

According to HTML5 spec, a button element cannot be inside an anchor tag. In fact, "but there must be no [interactive content](http://dev.w3.org/html5/spec/Overview.html#interactive-content) descendant

**Comment by shortcuts** [comment]
Thanks for reporting! Indeed we shouldn't nest anchors or buttons.

I wonder what a good alternative of the current display could be, I see absolute buttons being one but not sure if it's also accessible, definitely needs investigation. 

There is also other accessibility issues listed in #1370, tagging for visibility.

**Comment by brc-dd** [comment]
Yeah, I can't think of anything other than absolute buttons too. Another weird thing I noticed, Twitter is doing this:

```pug
div(role='option')
  Text here, dismiss button inside this:
  div(role='button') X
```

Though this isn't invalid HTML, but it is doing the same thing as DocSearch - both that option and button role divs are clickable.

**Comment by patrickhlauke** [comment]
As much as I hate those "ping" type comments ... wondering if there's any news/movement on this at all? Need any help on it?
